master/internal/api_checkpoint.go

```diff
@@ -28,6 +29,42 @@ func (a *apiServer) GetCheckpoint(
 	}
 }
 
+func (a *apiServer) CreateCheckpoint(
+	_ context.Context, req *apiv1.CreateCheckpointRequest,
+) (*apiv1.CreateCheckpointResponse, error) {
+	log.Infof("adding checkpoint %s (trial %d, batch %d)",
+		req.Checkpoint.Uuid, req.Checkpoint.TrialId, req.Checkpoint.BatchNumber)
+	modelC, err := model.CheckpointFromProto(req.Checkpoint)
+	if err != nil {
+		return nil, errors.Wrapf(
+			err, "error adding checkpoint %s (trial %d, batch %d) in database",
+			req.Checkpoint.Uuid, req.Checkpoint.TrialId, req.Checkpoint.BatchNumber)
+	}
+	err = a.m.db.AddCheckpoint(modelC)
+	return &apiv1.CreateCheckpointResponse{Checkpoint: req.Checkpoint},
+		errors.Wrapf(err, "error adding checkpoint %s (trial %d, batch %d) in database",
+			req.Checkpoint.Uuid, req.Checkpoint.TrialId, req.Checkpoint.BatchNumber)
```

Comment:
```suggestion
err = a.m.db.AddCheckpoint(modelC)
if err != nil {
	return nil, errors.Wrapf(err, "error adding checkpoint %s (trial %d, batch %d) in database",
		req.Checkpoint.Uuid, req.Checkpoint.TrialId, req.Checkpoint.BatchNumber)
}
return &apiv1.CreateCheckpointResponse{Checkpoint: req.Checkpoint}, nil
```
I have a preference for this so that resp is nil when there are errors, not just non-sense/corrupted data.

Reply:
fixed.